frontend/app/components/card.tsx

```diff
@@ -0,0 +1,161 @@
+import type { ComponentProps, HTMLElementType, KeyboardEvent, MouseEvent } from 'react';
+
+import type { To } from 'react-router';
+
+import type { FlipProp, IconProp } from '@fortawesome/fontawesome-svg-core';
+import { faChevronRight } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+import { AppLink } from '~/components/links';
+import type { I18nRouteFile } from '~/i18n-routes';
+import { cn } from '~/utils/tailwind-utils';
+
+type OptionalAppLink = Omit<ComponentProps<typeof AppLink>, 'to' | 'lang'> & {
+  to?: To;
+  file?: I18nRouteFile;
+};
```

Comment:
```suggestion
type OptionalAppLink = ComponentProps<typeof AppLink>;
```

**Reason:** `<AppLink>` already contains a `to`, and `file` prop. Omitting `to` and then adding it back via a type union is redundant. Also, as far as I can tell, there's no good reason to omit `lang`.

Reply:
I see the issue with my suggestion, as typescript gets angry because you can't spread `{...props}` into `<AppLink>`.

Gimme a bit to try something.

Reply:
Sorry.. what you did seems right. My bad. I'll resolve this comment, but I do have to make another one. on this line.